Include filename in reported errors

```scala
 echo "object a {" > foo.scala        
$ scalafmt -f foo.scala
[warn] Error in /Users/ollie/dev/scalafmt/target/foo.scala: <input>:2: error: } expected but end of file found

^
```

It would be nice if the error was reported like this instead
```scala
 echo "object a {" > foo.scala        
$ scalafmt -f foo.scala
/Users/ollie/dev/scalafmt/target/foo.scala:2: error: } expected but end of file found
object a {
         ^
```
like scalac does.

--- a/scalafmt-cli/src/main/scala/org/scalafmt/cli/Cli.scala
+++ b/scalafmt-cli/src/main/scala/org/scalafmt/cli/Cli.scala
@@ -13,6 +13,7 @@ import org.scalafmt.util.AbsoluteFile
 import org.scalafmt.util.FileOps
 import org.scalafmt.util.LogLevel
 import scala.meta.internal.tokenizers.PlatformTokenizerCache
+import scala.meta.parsers.ParseException
 
 object Cli {
   def nailMain(nGContext: NGContext): Unit = {
@@ -105,8 +106,12 @@ object Cli {
       inputMethod: InputMethod,
       options: CliOptions): Unit = {
     val input = inputMethod.readInput(options)
-    val formatResult =
-      Scalafmt.format(input, options.config, options.range)
+    val formatResult = Scalafmt.format(
+      input,
+      options.config,
+      options.range,
+      inputMethod.filename
+    )
     formatResult match {
       case Formatted.Success(formatted) =>
         inputMethod.write(formatted, input, options)
@@ -116,9 +121,14 @@ object Cli {
         } else if (options.config.runner.ignoreWarnings) {
           // do nothing
         } else {
-          options.common.err.println(
-            s"${LogLevel.warn} Error in ${inputMethod.filename}: $e"
-          )
+          e match {
+            case e: ParseException =>
+              options.common.err.println(e.toString())
+            case _ =>
+              options.common.err.println(
+                s"${LogLevel.warn} Error in ${inputMethod.filename}: $e"
+              )
+          }
         }
     }
   }

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala
@@ -33,8 +33,10 @@ object Scalafmt {
     */
   def format(
       code: String,
-      style: ScalafmtConfig = ScalafmtConfig.default,
-      range: Set[Range] = Set.empty[Range]): Formatted = {
+      style: ScalafmtConfig,
+      range: Set[Range],
+      filename: String
+  ): Formatted = {
     try {
       val runner = style.runner
       if (code.matches("\\s*")) Formatted.Success(System.lineSeparator())
@@ -45,7 +47,7 @@ object Scalafmt {
         } else {
           code
         }
-        val toParse = Rewrite(Input.String(unixCode), style)
+        val toParse = Rewrite(Input.VirtualFile(filename, unixCode), style)
         val tree = runner.dialect(toParse).parse(runner.parser).get
         val formatOps = new FormatOps(tree, style)
         runner.eventCallback(CreateFormatOps(formatOps))
@@ -71,6 +73,13 @@ object Scalafmt {
       // TODO(olafur) add more fine grained errors.
       case NonFatal(e) => Formatted.Failure(e)
     }
+  }
+
+  def format(
+      code: String,
+      style: ScalafmtConfig = ScalafmtConfig.default,
+      range: Set[Range] = Set.empty[Range]): Formatted = {
+    format(code, style, range, "<input>")
   }
 
   def parseHoconConfig(configString: String): Configured[ScalafmtConfig] =

--- a/scalafmt-tests/src/test/scala/org/scalafmt/cli/CliTest.scala
+++ b/scalafmt-tests/src/test/scala/org/scalafmt/cli/CliTest.scala
@@ -38,11 +38,15 @@ abstract class AbstractCliTest extends FunSuite with DiffAssertions {
 
   def getMockOptions(
       baseDir: AbsoluteFile,
-      workingDir: AbsoluteFile): CliOptions = {
+      workingDir: AbsoluteFile,
+      out: PrintStream = System.out
+  ): CliOptions = {
     CliOptions.default.copy(
-      gitOpsConstructor = x => new FakeGitOps(baseDir),
+      gitOpsConstructor = _ => new FakeGitOps(baseDir),
       common = CliOptions.default.common.copy(
-        workingDirectory = workingDir
+        workingDirectory = workingDir,
+        out = out,
+        err = out
       )
     )
   }
@@ -94,15 +98,19 @@ abstract class AbstractCliTest extends FunSuite with DiffAssertions {
   def noArgTest(
       input: AbsoluteFile,
       expected: String,
-      cmds: Seq[Array[String]]): Unit = {
+      cmds: Seq[Array[String]],
+      assertOut: String => Unit = println
+  ): Unit = {
     cmds.foreach { args =>
-      val init: CliOptions = getMockOptions(input)
+      val out = new ByteArrayOutputStream()
+      val init: CliOptions = getMockOptions(input, input, new PrintStream(out))
       val config = Cli.getConfig(args, init).get
       Cli.run(config)
       val obtained = dir2string(input)
       assertNoDiff(obtained, expected)
       val configTest = Cli.getConfig(Array("--test"), init).get
       Cli.run(configTest)
+      assertOut(out.toString())
     }
 
   }
@@ -477,6 +485,26 @@ class CliTest extends AbstractCliTest {
     assertNoDiff(obtained3, formatted)
   }
 
+  test("parse error is formatted nicely") {
+    val input =
+      """|/foo.scala
+         |object    A { foo( }
+         |""".stripMargin
+    noArgTest(
+      string2dir(input),
+      input,
+      Seq(Array.empty),
+      assertOut = out => {
+        assert(
+          out.contains(
+            """foo.scala:1: error: illegal start of simple expression
+              |object    A { foo( }
+              |                   ^""".stripMargin
+          )
+        )
+      }
+    )
+  }
   test("exception is thrown on invalid .scalafmt.conf") {
     val root = string2dir(
       """/.scalafmt.conf

